pkg/loadbalance/weighted_random_with_alias_method.go

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2023 CloudWeGo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package loadbalance
+
+import (
+	"context"
+
+	"github.com/bytedance/gopkg/lang/fastrand"
+	"github.com/cloudwego/kitex/pkg/discovery"
+)
+
+type AliseMethodPicker struct {
+	instances []discovery.Instance
+	weightSum int
+	alise     []int
+	prob      []float64
+}
+
+func newAliasMethodPicker(instances []discovery.Instance, weightSum int) Picker {
+	picker := &AliseMethodPicker{
+		instances: instances,
+		weightSum: weightSum,
+	}
+	picker.init()
+	return picker
+}
+
+// Alise Method need to init before use and after update instances
+func (a *AliseMethodPicker) init() {
+	n := len(a.instances)
+	a.alise = make([]int, n)
+	a.prob = make([]float64, n)
+
+	totalWeight := a.weightSum
+
+	scaledProb := make([]float64, n)
+	small := make([]int, 0)
```

Comment:
prealloc cap

Reply:
But the size of `small` and `large` is not certain at the begin

Reply:
There is a predictable cap upper bound `n`